### ISSUE TITLE
Using SVGPs with correlated heteroscedastic input uncertainties

### ISSUE THREAD

**Issue by cisprague** [cornellius-gp/gpytorch#1357]
I am trying to model a large-scale 2D field with an SVGP `f: (x,y) -> z`, where each input `(x,y)_i` has it's own distribution `N(0, Σ_i)`. Further, the input distributions are correlated with each other since they come from vehicle dead-reckoning. In this case, would there be an issue with using the input distribution sampling method shown in #913? Are there other methods that could accommodate this particular case? Could you recommend some things to look at in the literature for this? Thank you!

**Comment by jacobrgardner** [comment]
Are all input distributions correlated with all other input distributions? If so, you may have to ignore these cross input correlations for minibatch training to be technically correct.

If you do that, you can probably get away with the instance sampling from that example, or using a distributional kernel like here: https://github.com/cornellius-gp/gpytorch/blob/master/gpytorch/kernels/distributional_input_kernel.py. Basically the distributional kernel lets you pack the full distribution into the last dimension of `x` however you'd like, as long as you specify a distance function that computes a distribution distance between pairs of distributions.

**Comment by ignaciotb** [comment]
Thanks for the answer @jacobrgardner. All our inputs are measurements taken from a mobile robot, and so they're all correlated through the uncertainty in the robot localization. Would it be possible to use a "kernel mean embedding" kernel with the SVGP to take care of that?

**Comment by cisprague** [comment]
@jacobrgardner 
> Are all input distributions correlated with all other input distributions?

Indeed, all the inputs distributions are correlated with each other because they are determined through a Kálmán filter and a sensor measurement model. A valid approach we see is in [here](https://journals.sagepub.com/doi/abs/10.1177/0278364911421039), where they approximate a kernel over correlated distributions (Eq. 16) with a quadrature in Sec. 3.6.1. However, it's not a variational approach and doesn't scale.

We're also considering using the [variational GP-LVM](https://jmlr.org/papers/volume17/damianou16a/damianou16a.pdf) approach, but it's not clear to us how to admit correlated input distributions to this. :fearful:

**Comment by neildhir** [comment]
@cisprague did you find a suitable conclusion to this question? I am sitting in the same boat and would be keen to know what you went with. Thanks.
